### ISSUE TITLE
UnhandledPromiseRejectionWarning: FetchError on v1.28.1

### ISSUE THREAD

**Issue by HeritagePalms** [supabase/supabase-js#306]
# Bug report
Error to login using auth on 1.28.1 but not on 1.28.0

## Describe the bug
I get the following error after upgrade and trying to login:
UnhandledPromiseRejectionWarning: FetchError: invalid json response body at https://stuff.supabase.co/auth/v1/token?grant_type=password reason: Unexpected token < in JSON at position 0.

## Expected behavior
Not got this error message, which doesn't allow a user to login.

## System information

- OS: macOS
- Browser: Firefox, Chrome, Safari
- Version of supabase-js: 1.28.1
- Version of Node.js: 14.17.0

**Comment by soedirgo** [comment]
Hey @HeritagePalms, I wasn't able to reproduce this. What I tried:
```js
const { SupabaseClient } = require("@supabase/supabase-js")
const supabase = new SupabaseClient(
    "https://<...>.supabase.co",
    "<...>",
)
supabase.auth.signIn({ email: "<...>", password: "<...>" }).then(console.log)
```

Can you provide more info about your environment (e.g. Node.js/browser/service workers/Cloudflare Workers) and some repro code?

**Comment by HeritagePalms** [comment]
Thanks @soedirgo , Here is my basic setup.

`import supabase from '$lib/SupaBase/db';

export async function post(request) {
  let email = request.body.get('username')
  let password = request.body.get('password')

  const {error} = await supabase.auth.signIn({email, password})

  if (error) {
    return {
      status: error.status,
      body: error.message
    }
  }
  return {stuff...}
}`

I'm using Node 14.17.0
I tried this on Firefox, Safari & Chrome
I'm doing this locally. The code isn't public so I don't have a repo available.

**Comment by HeritagePalms** [comment]
I decided to update to 1.28.1 again after updating PostCSS & SvelteKit and my error went away...¯\_(ツ)_/¯

I don't know how they are correlated though so all I can share is that I did the following, in case it helps someone.
PostCSS: 8.3.11 -> 8.4.0
SvelteKit: 1-next.198 -> 1-next.199

**Comment by noskovvkirill** [comment]
Hi! I just had the same issue using "@supabase/supabase-js": "^1.28.1" with Next.js 
I had to update it to fix the error. 
The thing to worry about: while the error first appeared locally, all my production builds started to throw a server-side error. Once updated locally, production builds started to respond to the requests 🤷🏻 
While updating the client, I also restarted the supabase server. I'm not sure which one of those is responsible for the fix.
